src/careamics/config/data/ng_data_config.py

```diff
@@ -661,6 +780,212 @@ def set_means_and_stds(
+        field depending on the new mode.
+
+        To create a new training configuration, please use
+        `careamics.config.create_ng_data_configuration`.
+
+        This method compares the new parameters with the current ones and raises
+        errors if incompatible changes are requested, such as switching between 2D and
+        3D axes, or changing the number of channels. Incompatibility across parameters
+        may be delegated to Pydantic validation.
+
+        Parameters
+        ----------
+        new_mode : Literal["validating", "predicting"]
+            The new dataset mode, one of `validating` or `predicting`.
+        new_patching_strategy : PatchingConfig, default=None
+            New patching strategy to use. If None and switching to
+            `predicting`, the `whole` image strategy will be used. If swtiching to
+            `validating`, only `fixed_random` is supported.
+        new_batch_size : int, default=None
+            New batch size.
+        new_data_type : Literal['array', 'tiff', 'zarr', 'czi', 'custom'], default=None
+            New data type.
+        new_axes : str, default=None
+            New axes.
+        new_channels : Sequence of int, default=None
+            New channels.
+        new_in_memory : bool, default=None
+            New in_memory value.
+        new_dataloader_params : dict of {str: Any}, default=None
+            New dataloader parameters. These will be placed in the
+            appropriate dataloader params field depending on the new mode.
+
+        Returns
+        -------
+        NGDataConfig
+            New NGDataConfig with the updated mode and parameters.
+
+        Raises
+        ------
+        ValueError
+            If conversion to training mode is requested, or if incompatible changes
+            are requested.
+        """
+        if new_mode == Mode.TRAINING:
+            raise ValueError(
+                "Converting to 'training' mode is not supported. Create a new "
+                "NGDataConfig instead, for instance using "
+                "`create_ng_data_configuration`."
+            )
+        if self.mode != Mode.TRAINING:
+            raise ValueError(
+                f"Converting from mode '{self.mode}' to '{new_mode}' is not supported. "
+                f"Only conversion from 'training' mode is supported."
+            )
+
+        # sanity checks
+        # switching spatial axes
+        if new_axes is not None and ("Z" in new_axes) != (
+            "Z" in self.axes
+        ):  # switching 2D/3D
+            raise ValueError("Cannot switch between 2D and 3D axes.")
+
+        # switching channels
+        if (
+            new_axes is not None
+            and "C" in new_axes
+            and "C" not in self.axes  # adding C axis
+            and (
+                (new_channels is not None) and (len(new_channels) != 1)
+            )  # > 1 channels
+        ):
```

Comment:
Do you think this is clearer? Ignore if not

```suggestion
        adding_C_axis = (
            new_axes is not None and "C" in new_axes and "C" not in self.axes
        )
        selecting_singleton_C = (new_channels is not None) and (len(new_channels) == 1)
        if adding_C_axis and not selecting_singleton_C:
            ...
```

Reply:
Yes, it's much better! Thanks!!